File: Day-7/DivideStrings.java
class DivideStrings {
    public String[] divideString(String s, int k, char fill) {
        int n = s.length();
        int totalGroups = (int) Math.ceil((double) n / k);
        String[] result = new String[totalGroups];

        for (int i = 0; i < totalGroups; i++) {
            int start = i * k;
            int end = Math.min(start + k, n);
            String group = s.substring(start, end);

            // If group length is less than k, add fill characters
            if (group.length() < k) {
                group += String.valueOf(fill).repeat(k - group.length());
            }

            result[i] = group;
        }

        return result;
    }
}

File: Day-7/DivideStrings.go
import (
	"strings"
)

func divideString(s string, k int, fill rune) []string {
	n := len(s)
	var result []string
	for i := 0; i < n; i += k {
			end := i + k
			if end > n {
					result = append(result, s[i:] + strings.Repeat(string(fill), end - n))
			} else {
					result = append(result, s[i:end])
			}
	}
	return result
}